### PR TITLE
React Capstone Project final

### PR DESCRIPTION
**Pull request by JakariaJishan**
### Anime list
[video presentation](https://drive.google.com/file/d/1MavnfVoqiW9MxGEeSftHyr4wZGtvvlAm/view?usp=share_link)

When the page loads, the webapp shows the list of animes that could be filtered out by name or symbol.
Display some numeric values that come from the API.
The webapp retrieves data from the API to show detailed data in the category in the details page.
Creation of unit tests for pure functions in the Redux code.
Creation of integration tests using the React Testing Library.
The project is a single-page application (SPA) built with React and Redux.
The data retrieved from the API is stored in the Redux store.
filter the data you retrieve from the API using a Filter stateless component.
Every page (the main page and the pages for each item) has a unique route within the SPA.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,15 +70,17 @@
 
 ### Key Features <a name="key-features"></a>
 
-- **Calculator is currently on construction so it does not have features yet.**
+- **Anime List is currently on construction so it does not have features yet.**
```
Comment (alexander16108) [inline README.md:73]: as required your readme should contains an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

File: src/redux/Anime-list/episodeSlice.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(fetchEpisode.fulfilled, (state, action) => {
+      state.episodeList = action.payload;
+    });
+  },
+});
+
+export const { increment, decrement, incrementByAmount } = episodeSlice.actions;
```
Comment (alexander16108) [inline src/redux/Anime-list/episodeSlice.js]: in your live version, some of the items in the details are not able to fetch data from the API  which in turn does not render anything on the screen as seen below. it occurs across several animes please fix this 😄 
image

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @JakariaJishan , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__test__/Details.test.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  it('expects to initially set anime as an empty object', () => {
+    const state = store.getState().animeList;
+    const initialState = {
+      animeList: [],
+      loading: false,
+    };
+    expect(state).toEqual(initialState);
+  });
+  it('expects to display one anime episode from mock store', () => {
+    const episodes = [
```
Comment (alexander16108) [inline src/__test__/Details.test.js]: as required you should test for several components in your app but you have only tested your details page I suggest you test your home page as well and test some pure functions as well 😄

**Review by alexander16108 — CHANGES_REQUESTED** [review]

**Review by GedeonTS — APPROVED** [review]
# STATUS: APPROVED :tada: :clap: :champagne: 
Hi @JakariaJishan ,

You are amazing :smiley: :clap: !  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

 

## Highlights :clap: :tophat: 
- No linter errors :+1: 
- Good use of Gitflow :+1: 
- Good job adding the missing tests :+1: 
- The app is working fine :+1: 
- Professional README file, containing the Acknowledgement section :+1: 
- Descriptive PR summary :+1: 
## Optional suggestions :small_airplane: 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**